test/python/transpiler/test_commutative_cancellation.py

```diff
@@ -81,6 +81,44 @@ def test_all_gates(self):
         expected.global_phase = 0.5
         self.assertEqual(expected, new_circuit)
 
+    def test_2pi_multiples(self):
+        """Test 2pi multiples are handled with the correct phase they introduce."""
```

Comment:
Should we also add tests when the angles are not exact multiples by $\pi$, but differ from these within `_CUTOFF_PRECISION`?